.mergify.yml

```
@@ -43,14 +31,91 @@ queue_rules:
           - and: # breakage fails like we thought
               - check-failure=breakage
               - label=proto:expect-breakage
-    merge_conditions:
+    merge_checks: &merge-checks
       - base=master
       # Require integration tests before merging only
       - or:
           - label=bypass:integration
           - check-success=integration-test-result
 
+queue_rules:
+  - name: high_priority_rebase_main
+    commit_message_template: *commit-message-template
+    queue_conditions: *high-priority-queue-checks
+    merge_conditions: *merge-checks
+    merge_method: merge
+    update_method: rebase
+
+  - name: high_priority_merge_main
+    commit_message_template: *commit-message-template
+    queue_conditions: *high-priority-queue-checks
+    merge_conditions: *merge-checks
+    disallow_checks_interruption_from_queues:
+      - high_priority_rebase_main
```

Would it hurt to reference all queues in each queue of the "same priority" ?

```suggestion
    disallow_checks_interruption_from_queues:
      - high_priority_rebase_main
      - high_priority_merge_main
      - high_priority_squash_main
```

---

it wouldn't i think but its unnecessary since lower priority queues already don't preempt higher queues, and a queue wont preempt itself either